addons/mail/static/tests/mail_utils_tests.js

```diff
@@ -35,6 +35,13 @@ QUnit.test('add_link utility function', function (assert) {
     });
 });
 
+QUnit.test('add_link utility function and icons', function (assert) {
+    assert.expect(1);
+    // 'https://test.example.com/test?&currency_id' without escaping, the innerHTML from the browser convert into 'https://test.example.com/test?¤cy_id'
+    var output = utils.parseAndTransform('<p>https://test.example.com/test?&amp;currency_id</p>', utils.addLink);
```

Comment:
I tried to see how it would go with ```<p>https://test.example.com/test?&currency_id</p>```
and the result is actually broken the same, with `¤cy_id` after the fix.

So the fix is not correct, and a test should be added to cover that case too.

Reply:
i'm not sure this is an issue, `<p>https://test.example.com/test?&currency_id</p>` is equivalent to `<p>https://test.example.com/test?¤cy_id</p>` in HTML so your use case looks like expected behavior